components/ws-manager-bridge/src/bridge.ts

```diff
@@ -205,16 +205,15 @@ export class WorkspaceManagerBridge implements Disposable {
             );
             log.error(logCtx, "Failed to complete WorkspaceInstance status update", e);
             throw e;
-        } finally {
-            const durationMs = performance.now() - start;
-            this.prometheusExporter.reportWorkspaceInstanceUpdateCompleted(
-                durationMs / 1000,
-                writeToDB,
-                this.cluster.name,
-                status.spec.type,
-            );
-            log.info(logCtx, "Successfully completed WorkspaceInstance status update");
         }
+        const durationMs = performance.now() - start;
+        this.prometheusExporter.reportWorkspaceInstanceUpdateCompleted(
+            durationMs / 1000,
+            writeToDB,
+            this.cluster.name,
+            status.spec.type,
+        );
+        log.info(logCtx, "Successfully completed WorkspaceInstance status update");
```

Comment:
Alternatively you could bind `e` inside the catch to a variable outside of the try scope and then access it inside finally. This would avoid (almost) duplicating the report call.